shared/actions/wallets.tsx

```diff
@@ -1194,12 +1255,99 @@ const searchTrustlineAssets = (state, {payload: {text}}) => {
           WalletsGen.createSetTrustlineSearchResults({
-            assets: assets ? assets.map(rpcAsset => rpcAssetToAssetDescription(rpcAsset)) : [],
+            assets: assets
+              ? (assets
+                  .map(rpcAsset => rpcAssetToAssetDescriptionOrNative(rpcAsset))
+                  .filter(asset => asset !== 'native') as Array<Types.AssetDescription>)
+              : [],
           })
       )
     : WalletsGen.createClearTrustlineSearchResults()
 }
 
+const paymentPathToRpcPaymentPath = (paymentPath: Types.PaymentPath): RPCStellarTypes.PaymentPath => ({
+  destinationAmount: paymentPath.destinationAmount,
+  destinationAsset: assetDescriptionOrNativeToRpcAsset(paymentPath.destinationAsset),
+  path: paymentPath.path.toArray().map(ad => assetDescriptionOrNativeToRpcAsset(ad)),
+  sourceAmount: paymentPath.sourceAmount,
+  sourceAmountMax: paymentPath.sourceAmountMax,
+  sourceAsset: assetDescriptionOrNativeToRpcAsset(paymentPath.sourceAsset),
+})
+
+const rpcPaymentPathToPaymentPath = (rpcPaymentPath: RPCStellarTypes.PaymentPath) =>
+  Constants.makePaymentPath({
+    destinationAmount: rpcPaymentPath.destinationAmount,
+    destinationAsset: rpcAssetToAssetDescriptionOrNative(rpcPaymentPath.destinationAsset),
+    path: I.List(
+      rpcPaymentPath.path
+        ? rpcPaymentPath.path.map(rpcAsset => rpcAssetToAssetDescriptionOrNative(rpcAsset))
+        : []
+    ),
+    sourceAmount: rpcPaymentPath.sourceAmount,
+    sourceAmountMax: rpcPaymentPath.sourceAmountMax,
+    sourceAsset: rpcAssetToAssetDescriptionOrNative(rpcPaymentPath.sourceAsset),
+  })
+
+const calculateBuildingAdvanced = state =>
+  RPCStellarTypes.localFindPaymentPathLocalRpcPromise(
+    {
+      amount: state.wallets.buildingAdvanced.recipientAmount,
+      destinationAsset: assetDescriptionOrNativeToRpcAsset(state.wallets.buildingAdvanced.recipientAsset),
+      from: state.wallets.buildingAdvanced.senderAccountID,
+      sourceAsset: assetDescriptionOrNativeToRpcAsset(state.wallets.buildingAdvanced.senderAsset),
+      to: state.wallets.buildingAdvanced.recipient,
+    },
+    Constants.calculateBuildingAdvancedWaitingKey
+  )
+    .then(res => {
+      console.log({res})
+      const {
+        destinationAccount,
+        destinationDisplay,
+        exchangeRate,
+        fullPath,
+        sourceDisplay,
+        sourceMaxDisplay,
+      } = res
+      return WalletsGen.createSetBuiltPaymentAdvanced({
+        builtPaymentAdvanced: Constants.makeBuiltPaymentAdvanced({
+          amountError: 'You only have 34.04 EUR/Stonghold.com available to spend.',
+          destinationAccount,
+          destinationDisplay,
+          exchangeRate,
+          fullPath: rpcPaymentPathToPaymentPath(fullPath),
+          noPathFoundError: false,
+          readyToSend: true,
+          sourceDisplay,
+          sourceMaxDisplay,
+        }),
+      })
+    })
+    .catch(error => {
+      if (error && error.desc === 'no payment path found') {
```

Comment:
is it possible to check error.code instead of error.desc here?

Reply:
I think `error.code` here is just `218` which is generic error (will make sure). It'd be nice to turn this into a real error code in the future though. We have a bunch of them in Files too.